app/scripts/move-popup/dimMoveItemProperties.directive.js

```diff
@@ -165,12 +162,17 @@
       if (vm.item.primStat.statHash === 3897883278) {
         // it's armor.
         vm.light += ' Defense';
-        vm.classType = vm.item.classTypeName[0].toUpperCase() + vm.item.classTypeName.slice(1);
       } else if (vm.item.primStat.statHash === 368428387) {
         // it's a weapon.
         vm.light += ' Attack';
         vm.classes['is-' + vm.item.dmg] = true;
       }
+      if (vm.item.classTypeName !== 'unknown' &&
+          // These already include the class name
+          vm.item.type !== 'ClassItem' &&
+          vm.item.type !== 'Artifact') {
+        vm.classType = vm.item.classTypeName[0].toUpperCase() + vm.item.classTypeName.slice(1);
```

Comment:
why doesn't JS/es6 have a `.toCapitalCase` or something :( shame extending the string prototype is not an ideal solution... and is actually quite slow to boot. Maybe due to supporting multi locale? :/

![](http://i.stack.imgur.com/tNwKk.png)